lib/Dialect/SV/Transforms/RTLGreyBoxer.cpp

```diff
@@ -0,0 +1,64 @@
+//===- RTLCleanup.cpp - RTL Cleanup Pass ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This transformation pass converts external modules to empty normal modules.
+//
+//===----------------------------------------------------------------------===//
+
+#include "SVPassDetail.h"
+#include "circt/Dialect/SV/SVPasses.h"
+#include "circt/Support/ImplicitLocOpBuilder.h"
+
+using namespace circt;
+
+//===----------------------------------------------------------------------===//
+// GreyBoxer Pass
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct RTLGreyBoxerPass : public sv::RTLGreyBoxerBase<RTLGreyBoxerPass> {
+  void runOnOperation() override;
+};
+} // end anonymous namespace
+
+void RTLGreyBoxerPass::runOnOperation() {
+  auto topModule = getOperation().getBody();
+  SmallVector<rtl::RTLModuleExternOp, 8> toErase;
+  OpBuilder builder(topModule->getTerminator());
+
+  for (auto &op : *topModule)
```

Comment:
I think you can use `llvm::make_early_inc_range(*topModule)` and then erase the modules as you go.

Reply:
+1